crates/trie/common/src/proofs.rs

```diff
@@ -17,8 +17,27 @@ use alloy_trie::{
 use itertools::Itertools;
 use reth_primitives_traits::Account;
 
-/// Proof targets map.
-pub type MultiProofTargets = B256HashMap<B256HashSet>;
+/// Proof targets.
```

Comment:
I would massively benefit from an explainer what this type actually is, like what does multi,proof, and target mean

Reply:
added more context